### PR TITLE
GradleUp site

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Started a very basic prototype of the website, based on the Material for Mkdocs and the template. I have not yet updated graphics and styles, so it does not look very great

## Preview

https://oleg-nenashev.github.io/gradleup.github.io/

## Initial Content

- [x] Project pages, included from the repositories
- [x] Implement proper styles for the site
- [x] Proper logo 
- [x] Better landing page
- [x] Guidelines for joining the community channels

### CODE REVIEW

File: projects/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  A Gradle plugin to easily apply the same setup of static analysis tools
+  across different Android, Java or Kotlin projects.
+- [Gr8](../gr8/README.md) -
+  Gr8 makes it easy to shadow, shrink, and minimize your jars.
+
+## Gradle DevX
+
+- [Gratatouille](../gratatouille/README.md) -
+  Gratatouille is an opinionated framework to build Gradle plugins. Write pure Kotlin functions and the Gratatouille KSP processor generates tasks, workers and wiring code for you.
+- [New Maven Central Publishing](../nmcp/README.md) -
```
Comment (martinbonnin) [inline projects/README.md]: Nit I usually refer to it as just "nmcp", I find it shorter and easier to remember:

```suggestion
- [NMCP](../nmcp/README.md) (New Maven Central Publishing) -
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+* @oleg-nenashev
```
Comment (martinbonnin) [inline .github/CODEOWNERS]: I just created @gradleup/admins. Can we use that instead?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<p align="center">
+    <img width="512px" src="./assets/images/logo-wide.png" alt="WireMock Logo"/>
+</p>
+
+## What is GradleUp?
+
+GradleUp is an umbrella [GitHub organization](https://github.com/gradleup)
+focusing on simplifying Java and Android development with Gradle.
+On our menu: Gradle plugins, developer tools and best practices.
```
Comment (martinbonnin) [inline README.md]:
```suggestion
GradleUp is an umbrella [GitHub organization](https://github.com/gradleup) created by, and for, the community.
GradleUp focuses on simplifying building Java, Kotlin and Android projects.
On our menu: Gradle plugins, developer tools and best practices.
```

File: mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,159 @@
+site_name: GradleUp
+site_description: >-
+  GradleUp is an umbrella GitHub organization focusing
+  on simplifying Java and Android development with Gradle.
+  On our menu: Gradle plugins, developer tools and best practices.
+copyright: >-
+  Copyright &copy; 2024 - Oleg Nenashev and all GradleUp contributors.
```
Comment (martinbonnin) [inline mkdocs.yml]: Can we just go with "GradleUp contributors"?

```suggestion
  Copyright &copy; 2024 - GradleUp contributors.
```

File: docs/community/team.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# GradleUp Team
+
+The project is led by a small team,
+with a consensus-based decision making and governance.
+There is no special governance charter at the moment.
```
Comment (martinbonnin) [inline docs/community/team.md]: Nit: I would focus on the cool stuff more than what we don't have (yet!). Maybe something like so?

```suggestion
The project is led by a small team of passionate Gradle users and contributors.
The team is decentralized and decisions are made based on consensus and common sense.
```

File: docs/community/team.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+- [Alex Nordlund (@deepy)](https://github.com/deepy)
+- [Jean-Michel Fayard (@jmfayard)](https://github.com/jmfayard)
+- [Louis CAD (@LouisCAD)](https://github.com/LouisCAD)
+- [Martin Bonnin (@martinbonnin)](https://github.com/martinbonnin)
+- [Pablisco (@pablisco)](https://github.com/pablisco)
+- [Said Tahsin Dane (@tasomaniac)](https://github.com/tasomaniac)
+
+## Contact the Team
+
+For now, use DMs on the Kotlin Foundation or the Gradle Community Slack.
```
Comment (martinbonnin) [inline docs/community/team.md:18]: Would be cool to have a public `#gradleup`  on the Gradle slack

Comment (oleg-nenashev) [reply docs/community/team.md:18]: We can create one once we get the trademark usage grant. But yes, this is my assumption too

File: mkdocs.yml (hunk 4 above)
Comment (oleg-nenashev) [reply mkdocs.yml]: Fine by me

File: .github/CODEOWNERS (hunk 2 above)
Comment (oleg-nenashev) [reply .github/CODEOWNERS]:
```suggestion
* @GradleUp/admins
```

File: mkdocs.yml [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    - navigation.tabs
+    # - navigation.tabs.sticky
+    - navigation.top
+    - navigation.tracking
+    - search.highlight
+    - search.share
+    - search.suggest
+    - toc.follow
+    - toc.integrate
+  font:
```
Comment (oleg-nenashev) [inline mkdocs.yml]:
```suggestion
    - toc.integrate
    palette:
    - scheme: default
      media: "(prefers-color-scheme: light)"
      primary: custom
      toggle:
        icon: material/brightness-7
        name: Switch to dark mode
    - scheme: slate
      media: "(prefers-color-scheme: dark)"
      primary: custom
      toggle:
        icon: material/brightness-4
        name: Switch to light mode
  font:
```

File: mkdocs.yml [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    - navigation.tabs
+    # - navigation.tabs.sticky
+    - navigation.top
+    - navigation.tracking
+    - search.highlight
+    - search.share
+    - search.suggest
+    - toc.follow
+    - toc.integrate
+    palette:
```
Comment (oleg-nenashev) [inline mkdocs.yml]:
```suggestion
  palette:
```

File: CNAME [hunk 9]
```diff
@@ -1 +0,0 @@
-www.gradleup.com
```
Comment (martinbonnin) [inline CNAME:1]: Was that needed for DNS verification maybe? Ping @deepy ?

File: docs/community/hosting.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+To request the transfer and initiate the discussion, contact the team.
+
+## Criteria
+
+The general criteria for hosting a project are:
+
+- The project has value to the ecosystem and fits into the GradleUp scope.
+- The project must be open source and licensed under one of the
+  [OSI-approved open source licenses](https://opensource.org/),
+  preferably the Apache License v2.
```
Comment (martinbonnin) [inline docs/community/hosting.md]: Out of curiosity, any reason to prefer Apache vs MIT?

File: docs/community/participate.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+!!! info "Coming Soon!"
+
+    Community Channels will be published soon, stay tuned.
+
+## Team Slack
+
+We have a private community Slack.
+If you are a maintainer or an active contributor,
+reach out to one of the maintainers to get added.
```
Comment (martinbonnin) [inline docs/community/participate.md]: Nit: I would remove this section. As an outside user, it makes me feel as if the community is gated (which it in fact isn't). Maintainers can already contact the team to get their projects added and we can add them at that time.

```suggestion
```

File: docs/community/team.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+The team is decentralized and decisions are made based on consensus and common sense.
+
+## Members
+
+- [Alex Nordlund (@deepy)](https://github.com/deepy)
+- [Jean-Michel Fayard (@jmfayard)](https://github.com/jmfayard)
+- [Louis CAD (@LouisCAD)](https://github.com/LouisCAD)
+- [Martin Bonnin (@martinbonnin)](https://github.com/martinbonnin)
+- [Pablisco (@pablisco)](https://github.com/pablisco)
+- [Said Tahsin Dane (@tasomaniac)](https://github.com/tasomaniac)
```
Comment (martinbonnin) [inline docs/community/team.md:13]: We're missing @cortinico and you here :)

File: projects/README.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+## Gradle DevX
+
+- [Gratatouille](../gratatouille/README.md) -
+  Gratatouille is an opinionated framework to build Gradle plugins. Write pure Kotlin functions and the Gratatouille KSP processor generates tasks, workers and wiring code for you.
+- [NMCP](../nmcp/README.md) (New Maven Central Publishing) -
+  A plugin that uses the new Central Portal publisher API to publish to Maven Central.
+- [Librarian](../librarian/README.md) -
+  Librarian is a library to help building Kotlin libraries.
+- [Maven Sympathy](../maven-sympathy/README.md) -
+  A small Gradle plugin that checks that your project dependencies play nice with Maven resolution strategy
```
Comment (martinbonnin) [inline projects/README.md]: I'm not sure I get the distinction between Android and Gradle DevX. 

You can use GR8 to shrink/relocate any jar, not only Android for an example. Maybe make a single flat list?

File: overrides/home.html [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+        text-align:center;
+      }
+    }
+  </style>
+
+  <section class="mdx-container">
+    <div class="md-grid md-typeset">
+      <div class="mdx-hero">
+        <div class="mdx-hero__image">
+          <img src="assets/images/logo-2.png" alt="" width="360" draggable="false">
```
Comment (deepy) [inline overrides/home.html]:
```suggestion
          <img src="assets/images/logo-2.png" alt="GradleUp logo" width="360" draggable="false">
```

**Review by deepy — COMMENTED** [review]
Looks good to me, though the contrast in the footer on the landing page is too low - but that's not a blocker and can be adjusted afterwards

File: CNAME (hunk 9 above)
Comment (deepy) [reply CNAME:1]: Yeah we need to re-add it before/after merge, it's used by GH to map the pages to the domain
We don't need to worry about the verification as the domain is verified by the GH organization, but I think it's omitted here to make the preview easier

File: docs/community/hosting.md (hunk 10 above)
Comment (deepy) [reply docs/community/hosting.md]: My personal preference is always MIT or BSD, but Apache v2's patent grant is supposedly very useful for safeguarding against legal actions

File: docs/community/team.md (hunk 12 above)
Comment (cortinico) [reply docs/community/team.md:13]: <3

**Comment by cortinico** [comment]
> Looks good to me, though the contrast in the footer on the landing page is too low - but that's not a blocker and can be adjusted afterwards

I had the same feeling that the palette needs to be tweaked a bit as the contrast between title and background in dark mode is not really high

File: mkdocs.yml [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+nav:
+  - Projects:
+    - All Projects: projects/README.md
+    - Tools for Android:
+      - Auto Manifest: auto-manifest/README.md
+      - Gradle static analysis plugin: static-analysis-plugin/README.md
+      - Gr8: gr8/README.md
+    - Gradle DevX:
+      - Gratatouille: gratatouille/README.md
+      - NMCP (New Maven Central Publishing): nmcp/README.md
```
Comment (cortinico) [inline mkdocs.yml]: more of a nit:

The readme title reads a bit odd to me: `Nmcp: New Maven Central Publishing (or New Maven Central Portal too!)`. I think we should be mindful of the README files of the various projects as they will end up on the website.

Also status badges/emojis are all in different places.

**Review by cortinico — COMMENTED** [review]
Amazing work!

File: CNAME (hunk 9 above)
Comment (oleg-nenashev) [reply CNAME:1]: It is no longer needed for modern GitHub Pages. It was a culprit for static sites long ago

File: docs/community/team.md (hunk 12 above)
Comment (oleg-nenashev) [reply docs/community/team.md:13]: I will add the Shadow folks in a separate patch

```suggestion
- [Said Tahsin Dane (@tasomaniac)](https://github.com/tasomaniac)
- [Nicola Corti (@cortinico)](https://github.com/cortinico)
```

File: docs/community/hosting.md (hunk 10 above)
Comment (oleg-nenashev) [reply docs/community/hosting.md]: Yep, Apache License prevents from patent claims. Will remove the specifics for now though

File: mkdocs.yml (hunk 15 above)
Comment (oleg-nenashev) [reply mkdocs.yml]: Yes, updating the downstream READMEs is definitely advised

File: projects/README.md (hunk 13 above)
Comment (oleg-nenashev) [reply projects/README.md]: For me it seemed to be too big for a single list. But yes, you are right, right now the classification is not very helpful. I will remove it, we can always restore it or do another one in the future

**Comment by oleg-nenashev** [comment]
Will merge as agreed on Slack

**Pull request merged**